### PR TITLE
Hot fix/us12181 logging for trips

### PR DESCRIPTION
**Pull request by IngageRitterS**
Adding logging for trips processing.  Helpdesk tickets are increasingly difficult to troubleshoot and we are hoping that the additional logging will enable su to track down and correct all trips issues.

### CODE REVIEW

File: Gateway/MinistryPlatform.Translation/Repositories/FormSubmissionRepository.cs [hunk 1]
```diff
@@ -160,15 +162,29 @@ private static IDbCommand CreateTripFormResponsesSqlCommandWithSelectionId(int s
...
-            return responseId;
+            catch
+            {
+                JsonSerializerSettings settings = new JsonSerializerSettings
+                {
+                    Error = (serializer, err) => err.ErrorContext.Handled = true
+                };
+                string json = JsonConvert.SerializeObject(form, settings);            
+                _logger.Error($"Failure in SubmitFormResponse: {json}");
+                throw new Exception();
```
Comment (chris-branch) [inline Gateway/MinistryPlatform.Translation/Repositories/FormSubmissionRepository.cs]: Should we re-throw the original exception vs. throwing a new exception ?

File: Gateway/MinistryPlatform.Translation/Repositories/TripRepository.cs [hunk 2]
```diff
@@ -25,37 +25,41 @@ public TripRepository(IMinistryPlatformRestRepository ministryPlatformRestReposi
 
         public Result<MpPledge> AddAsTripParticipant(int ContactId, int PledgeCampaignID, string token)
         {
-            var storedProc = _configurationWrapper.GetConfigValue("TripParticipantStoredProc");
+            var storedProc = _configurationWrapper.GetConfigValue("TripParticipantStoredProc"); 
+            _logger.Warn($"The stored proc read from the config is {storedProc}");
```
Comment (chris-branch) [inline Gateway/MinistryPlatform.Translation/Repositories/TripRepository.cs]: I'm not sure that prod has the log level set to show warnings, but we can discuss.

File: Gateway/MinistryPlatform.Translation/Repositories/TripRepository.cs [hunk 3]
```diff
@@ -25,37 +25,41 @@ public TripRepository(IMinistryPlatformRestRepository ministryPlatformRestReposi
...
                 }
-                _logger.Debug($"Adding a trip particpant returned no results. The trip is already full.");
+                _logger.Warn($"Adding a trip participant returned no results. The trip is already full.");
                 return new Result<MpPledge>(false, "Trip is already full");
             }
             catch (Exception e)
             {
-                _logger.Error($"Failed to call stored procedure #{storedProc}");
-                _logger.Error(e.Message);
+                _logger.Error($"Failed to call stored procedure #{storedProc} for contactId {ContactId} pledge campaign Id {PledgeCampaignID}");
```
Comment (chris-branch) [inline Gateway/MinistryPlatform.Translation/Repositories/TripRepository.cs]: Should we include the exception "e" in the Error() call so we get a call stack, etc. ?

File: Gateway/crds-angular/Services/TripService.cs [hunk 4]
```diff
@@ -516,17 +514,23 @@ public int SaveApplication(TripApplicationDto dto)
...
                     PledgeCampaignId = dto.PledgeCampaignId,
                     FormAnswers = new List<MpFormAnswer>(FormatFormAnswers(dto))
                 };
-                
+                JsonSerializerSettings settings = new JsonSerializerSettings
+                {
+                    Error = (serializer, err) => err.ErrorContext.Handled = true
+                };
+                string json = JsonConvert.SerializeObject(dto, settings);
+                _logger.Error($"Failure in SubmitFormResponse: {json}");
```
Comment (chris-branch) [inline Gateway/crds-angular/Services/TripService.cs]: Is this actually a failure, or just logging for info purposes ?

File: Gateway/MinistryPlatform.Translation/Repositories/FormSubmissionRepository.cs (hunk 1 above)
Comment (IngageRitterS) [reply Gateway/MinistryPlatform.Translation/Repositories/FormSubmissionRepository.cs]: Yes!

File: Gateway/MinistryPlatform.Translation/Repositories/TripRepository.cs [hunk 5]
```diff
@@ -45,10 +45,10 @@ public Result<MpPledge> AddAsTripParticipant(int ContactId, int PledgeCampaignID
                 _logger.Warn($"Adding a trip participant returned no results. The trip is already full.");
                 return new Result<MpPledge>(false, "Trip is already full");
             }
-            catch (Exception e)
+            catch (Exception ex)
             {
                 _logger.Error($"Failed to call stored procedure #{storedProc} for contactId {ContactId} pledge campaign Id {PledgeCampaignID}");
-                _logger.Error(e.Message);               
+                _logger.Error(ex, ex.Message);
```
Comment (chris-branch) [inline Gateway/MinistryPlatform.Translation/Repositories/TripRepository.cs]: I think the exception "ex" should be the 2nd param, so swap the order.

**Pull request merged**